packages/expo/test/index.test.ts

```diff
@@ -0,0 +1,163 @@
+  readAsStringAsync: jest.fn(() => Promise.resolve()),
+  writeAsStringAsync: jest.fn(() => Promise.resolve()),
+  deleteAsync: jest.fn(() => Promise.resolve()),
+  moveAsync: jest.fn(() => Promise.resolve()),
+  copyAsync: jest.fn(() => Promise.resolve()),
+  makeDirectoryAsync: jest.fn(() => Promise.resolve()),
+  readDirectoryAsync: jest.fn(() => Promise.resolve()),
+  createDownloadResumable: jest.fn(() => Promise.resolve())
+}))
+
+jest.mock('../../delivery-expo/node_modules/@react-native-community/netinfo', () => ({
+  addEventListener: jest.fn(),
+  fetch: () => Promise.resolve({ isConnected: true, dogger: 'tes' })
+}))
+
+jest.mock('../../plugin-react-native-connectivity-breadcrumbs/node_modules/@react-native-community/netinfo', () => ({
+  addEventListener: jest.fn(),
+  fetch: () => Promise.resolve({ isConnected: true })
+}))
+
+jest.doMock('../../plugin-expo-device/node_modules/expo-device', () => ({
+  manufacturer: 'Google',
+  modelName: 'Pixel 4'
+}))
+
+declare global {
+  namespace NodeJS { // eslint-disable-line
+    interface Global {
+      ErrorUtils: {
+        setGlobalHandler: (fn: Function) => void
+        getGlobalHandler: () => Function
+      }
+    }
+  }
+}
+
+global.ErrorUtils = {
+  setGlobalHandler: jest.fn(),
+  getGlobalHandler: jest.fn()
+}
+
+const API_KEY = '030bab153e7c2349be364d23b5ae93b5'
+
+describe('expo notifier', () => {
+  let Bugsnag: typeof BugsnagExpoStatic
+
+  beforeEach(() => {
+    jest.isolateModules(() => {
+      Bugsnag = require('..')
+    })
+  })
+
+  it('accepts plugins', () => {
+    Bugsnag.start({
+      apiKey: API_KEY,
+      plugins: [{
+        name: 'foobar',
+        load: client => 10
+      }]
+    })
+    expect(Bugsnag.getPlugin('foobar')).toBe(10)
+  })
+
+  it('notifies handled errors', () => {
+    // Explicitly reference the public types to ensure they are exported correctly
+    const error: NotifiableError = new Error('123')
+    const onError: OnErrorCallback = (event: Event) => {}
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const client: Client = Bugsnag.start({
+      apiKey: API_KEY,
+      appVersion: '1.2.3',
+      appType: 'worker',
+      autoDetectErrors: true,
+      enabledErrorTypes: {
+        unhandledExceptions: true,
+        unhandledRejections: true
+      },
+      onError: [
+        event => true
+      ],
+      onBreadcrumb: (b: Breadcrumb) => {
+        return false
+      },
+      onSession: (s: Session) => {
+        return true
+      },
+      endpoints: { notify: 'https://notify.bugsnag.com', sessions: 'https://sessions.bugsnag.com' },
+      autoTrackSessions: true,
+      enabledReleaseStages: ['production'],
+      releaseStage: 'production',
+      maxBreadcrumbs: 20,
+      enabledBreadcrumbTypes: ['manual', 'log', 'request'],
+      user: null,
+      metadata: {},
+      logger: undefined,
+      redactedKeys: ['foo', /bar/]
+    })
+
+    Bugsnag.notify(error, onError)
+    expect(true).toBe(true)
```

Comment:
I know a lot of it is mocked, but there must be a better assertion we can make about the client here?

Reply:
I'm struggling to know what to do here. I want to mock fetch and assert it was called. However, for reasons I don't fully understand, but I think is to do with the monorepo setup, I'm unable to mock the fetch that `delivery-expo` imports.

Reply:
Ok I've resorted to mocking delivery-expo and asserting on that. It's better. For some reason though, I cannot assert on the exceptions part of the payload. It's there in the jest diff but when I try to access `.exceptions`, it's undefined.

Reply:
ah, I did a little digging and this seems to be due to the fact that the `event` object at runtime has a `.errors` property, but to match the API, when this gets [JSON stringified it gets renamed to `exceptions`](https://github.com/bugsnag/bugsnag-js/blob/ac7172ad6f794e68833ca2cb22227ae585912abd/packages/core/event.js#L82).

So when Jest output the actual value when it doesn't match the expected value, it looks like it JSON-stringifies the actual value.

I added the following and it worked:

```diff
      expect(_delivery.sendEvent).toHaveBeenCalledWith(expect.objectContaining({
        apiKey: '030bab153e7c2349be364d23b5ae93b5',
        events: expect.arrayContaining([
          expect.objectContaining({
            app: expect.objectContaining({ releaseStage: 'production', version: '1.2.3' }),
            breadcrumbs: [],
            device: expect.objectContaining({ manufacturer: 'Google', model: 'Pixel 4', modelNumber: undefined, osName: 'android', totalMemory: undefined }),
+             errors: expect.arrayContaining([
+               expect.any(Object)
+            ])
          })
        ]),
        notifier: { name: 'Bugsnag Expo', url: 'https://github.com/bugsnag/bugsnag-js', version: '7.5.2' }
      }), expect.any(Function))
```